File: apps/miniprogram/src/api/comments.ts
import { request } from '../utils/request'

export interface CommentItem {
  id: number
  user_id: number
  song_id: number
  parent_id?: number | null
  content: string
  like_count: number
  created_at: string
}

export function getSongComments(songId: number) {
  return request<CommentItem[]>(`/comments/song/${songId}`, { auth: false })
}

export function createComment(songId: number, content: string, parentId?: number | null) {
  return request<CommentItem>('/comments', {
    method: 'POST',
    data: {
      song_id: songId,
      content,
      parent_id: parentId || null,
    },
  })
}

File: apps/server/app/api/v1/comments.py
from fastapi import APIRouter,Depends
from sqlalchemy.orm import Session
from app.api.deps import get_current_user
from app.core.database import get_db
from app.models.comment import Comment
from app.models.user import User
from app.schemas.comment import CommentCreateIn,CommentOut
from app.schemas.common import ApiResponse

router=APIRouter(prefix='/comments',tags=['comments'])

@router.post('',response_model=ApiResponse[CommentOut])
def create(payload:CommentCreateIn,db:Session=Depends(get_db),user:User=Depends(get_current_user)):
 c=Comment(user_id=user.id,song_id=payload.song_id,parent_id=payload.parent_id,content=payload.content)
 db.add(c)
 db.commit()
 db.refresh(c)
 return ApiResponse(data=CommentOut.model_validate(c))

@router.get('/song/{song_id}',response_model=ApiResponse[list])
def list_comments(song_id:int,db:Session=Depends(get_db)):
 items=db.query(Comment).filter(Comment.song_id==song_id).order_by(Comment.created_at.desc()).all()
 return ApiResponse(data=[CommentOut.model_validate(x).model_dump() for x in items])